src/compiler/checker.ts

```diff
@@ -6134,10 +6136,23 @@ module ts {
                     }
                     lastSymbol = symbol;
 
-                    for (var j = result.length; j > pos; j--) {
+                    // specialized signatures always need to be placed before non-specialized signatures regardless
+                    // of the cutoff position; see GH#1133
+                    if (signature.hasStringLiterals) {
+                        splicePos = ++specializedPos;
+                        // The cutoff position needs to be increased to account for the fact that we are adding things
+                        // before the cutoff point. If the cutoff position is not incremented, merged interfaces will
+                        // start adding their merged signatures at the wrong position
+                        ++cutoffPos;
+                    }
+                    else {
+                        splicePos = pos;
+                    }
+
+                    for (var j = result.length; j > splicePos; j--) {

```

Comment:
Yes please. I was just trying to touch as little as possible since it was my first patch. :)